src/main/java/org/opensearch/neuralsearch/processor/NormalizationProcessorWorkflow.java

```diff
@@ -57,44 +55,30 @@ public class NormalizationProcessorWorkflow {
 
     /**
      * Start execution of this workflow
-     * @param querySearchResults input data with QuerySearchResult from multiple shards
-     * @param normalizationTechnique technique for score normalization
-     * @param combinationTechnique technique for score combination
+     * @param request contains querySearchResults input data with QuerySearchResult
+     * from multiple shards, fetchSearchResultOptional, normalizationTechnique technique for score normalization
+     *  combinationTechnique technique for score combination, and nullable rankConstant only used in RRF technique
      */
-    public void execute(
-        final List<QuerySearchResult> querySearchResults,
-        final Optional<FetchSearchResult> fetchSearchResultOptional,
-        final ScoreNormalizationTechnique normalizationTechnique,
-        final ScoreCombinationTechnique combinationTechnique,
-        final SearchPhaseContext searchPhaseContext
-    ) {
-        NormalizationProcessorWorkflowExecuteRequest request = NormalizationProcessorWorkflowExecuteRequest.builder()
-            .querySearchResults(querySearchResults)
-            .fetchSearchResultOptional(fetchSearchResultOptional)
-            .normalizationTechnique(normalizationTechnique)
-            .combinationTechnique(combinationTechnique)
-            .explain(false)
-            .searchPhaseContext(searchPhaseContext)
-            .build();
-        execute(request);
-    }
-
     public void execute(final NormalizationProcessorWorkflowExecuteRequest request) {
         List<QuerySearchResult> querySearchResults = request.getQuerySearchResults();
         Optional<FetchSearchResult> fetchSearchResultOptional = request.getFetchSearchResultOptional();
-
-        // save original state
-        List<Integer> unprocessedDocIds = unprocessedDocIds(querySearchResults);
+        List<Integer> unprocessedDocIds = unprocessedDocIds(request.getQuerySearchResults());
```

Comment:
Why did we revert this change? We are already getting querySearchResults at line 63.  It saves the call of request.getQuerySearchResults() again

Reply:
I'm not sure, it's present in main too.  I can raise a PR to fix that in main first then backport

Reply:
I remember I made this change in main. Can you once check

Reply:
https://github.com/opensearch-project/neural-search/issues/1109

Reply:
Yeah I think the change you made was reverted when this PR was merged to main since this is just a backport.  I just double checked and the duplicate is there in main